scripts/ci/lib.sh

```diff
@@ -1119,14 +1119,6 @@ validate_expected_go_version() {
         echo "Got unexpected go version ${roxctl_go_version} (wanted ${expected_go_version})"
         exit 1
     fi
-
```

Comment:
I don't know why this was added, but I do not think it makes sense. The entry in `go.mod` is the source compatibility version, its only relation to the go version used for building is that the source compatibility level must not be above the builder go version - but in that case, builds would fail anyway.

Whether or not we leave this at 1.17 or set to 1.18 is more similar to whether or not we use generics in the code.

Reply:
Yes, I actually do not like this generally speaking